src/Internal/Service/OrderService.php

```diff
@@ -48,4 +92,74 @@ public function __construct( WC_Payments_Order_Service $legacy_service ) {
 	public function set_payment_method_id( int $order_id, string $payment_method_id ) {
 		$this->legacy_service->set_payment_method_id_for_order( $order_id, $payment_method_id );
 	}
+
+	/**
+	 * Generates payment metadata from order details.
+	 *
+	 * @param int          $order_id     ID of the order.
+	 * @param Payment_Type $payment_type Type of the payment (recurring or not).
+	 * @return array                     The metadat athat will be sent to the server.
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function get_payment_metadata( int $order_id, Payment_Type $payment_type = null ) {
+		$order = $this->get_order( $order_id );
+
+		$name     = sanitize_text_field( $order->get_billing_first_name() ) . ' ' . sanitize_text_field( $order->get_billing_last_name() );
+		$email    = sanitize_email( $order->get_billing_email() );
+		$metadata = [
```

Comment:
Oh, I think we missed one thing here. 
Should we add a new meta value to differentiate between the old processing and the new one, on which we're working here? 

Not necessarily to do that this PR, we can create an issue to enhance this later.

Reply:
It's not too late. I'll add this when fixing conflicts with the base PR.

Reply:
Actually, I'll do that shortly in https://github.com/Automattic/woocommerce-payments/pull/7332, as it needs to be done outside of the service.